### PR TITLE
set division operator call to return NaN on 0/0

### LINKED ISSUE #377: Not seeing a NaN when expected on initial import
_Original author: PaulMake...@gmail.com (May 10, 2011 17:48:18)_

<b>What steps will reproduce the problem?</b>
1. Import the attached file (containing simply the number 0 and a header)
2. Edit cells -&gt; Transform: value / value

<b>What is the expected output? What do you see instead?</b>

Expect to see NaN - see instead null.

<b>What version of Google Refine are you using?</b>

2.0

<b>What operating system and browser are you using?</b>

Windows Chrome

<b>Please provide any additional information below.</b>

If I transform the column to text and then back to numbers I get the expected NaN as though the initial import missed something in construction.

_Original issue: http://code.google.com/p/google-refine/issues/detail?id=377_

### PR DESCRIPTION
**Pull request by elroykanye**
Fixes #377

Changes proposed in this pull request:
- fix the division operator call to return a Double.NaN when `n1 <- 0, n2 <- 0`

### CODE REVIEW

File: main/src/com/google/refine/grel/ast/OperatorCallExpr.java [hunk 1]
```diff
@@ -101,6 +102,7 @@ public Object evaluate(Properties bindings) {
                     } else if ("*".equals(_op)) {
                         return n1 * n2;
                     } else if ("/".equals(_op)) {
+                        if(n2 == 0&& n1 == 0) { return Double.NaN; }
```
Comment (wetneb) [inline main/src/com/google/refine/grel/ast/OperatorCallExpr.java:105]:
```suggestion
                        if (n2 == 0 && n1 == 0) {
                            return Double.NaN;
                        }
```

**Review by wetneb — COMMENTED** [review]
Thank you! Could you add a unit test as well?

File: main/src/com/google/refine/grel/ast/OperatorCallExpr.java (hunk 1 above)
Comment (wetneb) [reply main/src/com/google/refine/grel/ast/OperatorCallExpr.java:105]: We normally have linting checks which should pick this up but it looks like they are not quite strict enough. You could do a similar change above.

Comment (elroykanye) [reply main/src/com/google/refine/grel/ast/OperatorCallExpr.java:105]: Alright, I will. Thanks! :)

**Review by wetneb — APPROVED** [review]
Thanks @elroykanye!

**Comment by elroykanye** [comment]
> Thanks @elroykanye!

My pleasure 😁

**Pull request merged**